cmd/cogito/main_test.go

```diff
@@ -11,21 +11,19 @@ import (
 
 func TestRunSmokeSuccess(t *testing.T) {
 	type testCase struct {
-		name    string
-		args    []string
-		in      string
-		wantOut string
+		name string
+		args []string
+		in   string
 	}
 
 	test := func(t *testing.T, tc testCase) {
 		in := strings.NewReader(tc.in)
 		var out bytes.Buffer
+		var logOut bytes.Buffer
 
-		err := run(in, &out, io.Discard, tc.args)
+		err := run(in, &out, &logOut, tc.args)
 
-		assert.NilError(t, err)
-		have := out.String()
-		assert.Equal(t, have, tc.wantOut)
+		assert.NilError(t, err, "\nout: %s\nlogOut: %s", out.String(), logOut.String())
```

Comment:
The majority of the [assert](https://pkg.go.dev/gotest.tools/v3/assert) functions have the last parameters of the signature the same:
```
func Equal(t testing.T, have, want interface{}, msgAndArgs ...interface{})
```
where the optional
```
msgAndArgs ...interface{}
```
are printed in case of failure, to give more context.